01-Login/README.md

```diff
@@ -1,3 +1,84 @@
 # Login
 
-[Full Tutorial](https://auth0.com/docs/quickstart/webapp/java-spring-security-mvc/01-login)
\ No newline at end of file
+[Full Tutorial](https://auth0.com/docs/quickstart/webapp/java-spring-security-mvc/01-login)
+
+## Getting started
+
+This sample demonstrates how to use Auth0 to perform authentication using the `mvc-auth-commons` library. Download or clone this repository and follow the instructions below to setup the sample.
+
+### Auth0 Dashboard
+1. On the [Auth0 Dashboard](https://manage.auth0.com/#/clients) create a new Client of type `Regular Web Application`.
+1. Add the URL that will be called on an OAuth successful login to the Allowed Callback URLs. i.e.: `https://mysite.com/callback`.
+1. Add the URL that will be called on logout to the Allowed Logout URLs. i.e.: `https://mysite.com/logout`.
+1. Copy the `Domain`, `Client ID` and `Client Secret` values at the top of the page and use them to configure the Java Application.
+
+
+### Java Application
+Set the client values in the `src/main/resources/auth0.properties` file. They are read by the `AppConfig` class.
+
+```xml
+com.auth0.domain: {YOUR_AUTH0_DOMAIN}
+com.auth0.clientId: {YOUR_AUTH0_CLIENT_ID}
+com.auth0.clientSecret: {YOUR_AUTH0_CLIENT_SECRET}
+```
+
+It will request by default a `code` Response Type and later execute a Code Exchange. You can modify this behavior by changing the Response Type in the `AuthController` class to `token` or `id_token` to use Implicit Grant. i.e.:
+
+```java
+AuthenticationController.newBuilder(domain, clientId, clientSecret)
+    .withResponseType("token")
+    .build();
+```
+
+Keep in mind that the server uses `POST` to return an Implicit Grant result, you should handle that on your controller too.
+
+
+### Running the sample
+
+Open a terminal, go to the project root directory and run the following command:
+
+```bash
+./gradlew clean bootRun
+```
+
+The server will be accessible on https://localhost:8080/portal/home. After logging in you should see the `token` in the header.
+
+### Running the sample with docker
+
+In order to run the example with docker you need to have `docker` installed.
```

Comment:
I'd add a link to the [installation guide](https://docs.docker.com/engine/installation/)